src/jvm/test/org/jetbrains/kotlinx/lincheck/test/GeneratorSeedTest.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2023 JetBrains s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>
+ */
+
+package org.jetbrains.kotlinx.lincheck.test
+
+import org.jetbrains.kotlinx.lincheck.LincheckAssertionError
+import org.jetbrains.kotlinx.lincheck.LoggingLevel
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import org.jetbrains.kotlinx.lincheck.annotations.Param
+import org.jetbrains.kotlinx.lincheck.check
+import org.jetbrains.kotlinx.lincheck.paramgen.IntGen
+import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.ModelCheckingOptions
+import org.junit.Test
+
+/**
+ * This test checks that parameter generators random use different seeds than executions generator.
+ * This test fails if seeds are equals.
+ * @see <a href="https://github.com/Kotlin/kotlinx-lincheck/issues/120">Corresponding bug description</a>.
+ */
+@Param(name = "key", gen = IntGen::class, conf = "0:1")
+class GeneratorSeedTest {
```

Comment:
You should also check that different values are generated when an operation has multiple arguments of the same type. The same if one of the arguments has `@Param` annotation, or both of them.